pandas/_typing.py

```diff
@@ -93,10 +93,21 @@
         from typing import Self
     else:
         from typing_extensions import Self  # pyright: reportUnusedImport = false
+
+    try:
+        from pyarrow import (
```

Comment:
IIUC we expect optional dependencies to be present for mypy checks, but not sure.  who would know?

Reply:
Yes, that's also fine with me, that would just simplify this a bit.